### PR TITLE
#63 fe create a light and dark theme dark theme should be default

### PR DESCRIPTION
**Pull request by DimitarHristovski**
I have created the Theme component using context.Let me know if theres something additional things to do.

### CODE REVIEW

File: app/styles/globals/_typography.scss [hunk 1]
```diff
@@ -90,3 +90,13 @@ h4 {
   line-height: 16px;
   letter-spacing: 0.4px;
 }
+.day {
+  background-color: #fff !important;
+  color: #000 !important;
+}
+
+.night {
```
Comment (edichoska) [inline app/styles/globals/_typography.scss:98]: use better naming for this classes

**Comment by edichoska** [comment]
Screenshot 2024-03-29 at 12 17 08
@DimitarHristovski  the implementation of the context api and the theme is great. 
Could you please fix the issues with the UI:
1.  It seems like the navigation has a background color, please remove it
2. The navigation looks broken. Please take your time and make sure it looks okay

File: app/page.tsx [hunk 2]
```diff
@@ -12,7 +12,6 @@ const Home = () => {
           заедница од ентузијасти и професионалци. Со LearnHub, вие не сте само дел од заедница; вие
           градите иднина."
       />
-      <BlogList />
```
Comment (edichoska) [inline app/page.tsx:15]: why did you remove this?

File: components/context/ThemeContext.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+interface ThemeContextType {
+  theme: 'day' | 'night';
+  toggleTheme: () => void;
+}
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+export const ThemeProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const [theme, setTheme] = useState<'day' | 'night'>('day');
```
Comment (edichoska) [inline components/context/ThemeContext.tsx:14]: the default theme should be dark.
please rename this to be light and dark, with dark being the default

File: app/layout.tsx [hunk 4]
```diff
@@ -14,15 +19,17 @@ const RootLayout = ({
...
+    <html lang="en">
+      <body className={roboto.className}>
+        <Navigation />
+        <main>
+          <Suspense fallback={<Loading />}>{children}</Suspense>
+        </main>
+        <Footer />
+      </body>
+    </html>
+  </ThemeProvider>
```
Comment (edichoska) [inline app/layout.tsx:32]: the <ThemeProvider> should be inside the body tag

File: components/context/ThemeContext.tsx (hunk 3 above)
Comment (edichoska) [reply components/context/ThemeContext.tsx:14]: Also we might need to use localStorage/cookies + useEffect to achieve this.

**Review by edichoska — CHANGES_REQUESTED** [review]

**Comment by stoimenov95** [comment]
@DimitarHristovski please use a shorter name for the branch the next time, including "-" for separating the words.

File: components/reusable-components/navigation/navigation.module.scss [hunk 5]
```diff
@@ -124,3 +124,24 @@
...
+.animate {
+  transition: transform 1.5s ease; 
+  display: flex;
+  align-content: center;
+}
+
+.moveRight {
+  transform: translateX(20px); 
+}
+.BKG{
```
Comment (stoimenov95) [inline components/reusable-components/navigation/navigation.module.scss:136]: @DimitarHristovski we can use better naming for these BKG classes as well

**Review by stoimenov95 — CHANGES_REQUESTED** [review]
A merge conflicts resolve is needed, and better class naming.

**Comment by stoimenov95** [comment]
@DimitarHristovski the theme should not update on page refresh.

**Comment by edichoska** [comment]
@DimitarHristovski why did you close this PR?

File: app/page.tsx (hunk 2 above)
Comment (DimitarHristovski) [reply app/page.tsx]: Oh it was causing conflicts in the begining... i believe because the final update of that branch wasnt pushed and the page wasnt rendering.When i merged the main with my branch everything was fixed with the latest updates of the other branches.But works fine.

**Comment by DimitarHristovski** [comment]
> @DimitarHristovski why did you close this PR?

When i fix everything and is ready for merge im gonna open it again

**Comment by edichoska** [comment]
> > @DimitarHristovski why did you close this PR?
> 
> When i fix everything and is ready for merge im gonna open it again now the ticked is back in progress

no, leave it. You can just return the ticket in In Progress

**Comment by DimitarHristovski** [comment]
> @DimitarHristovski how is it going with this ticket? Do you need some help with the changes we requested?

Can you please check what more i should change because i think its ready.

File: components/context/ThemeContext.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+interface ThemeContextType {
+  theme: 'light' | 'dark';
+  toggleTheme: () => void;
+}
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+export const ThemeProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const storedTheme = localStorage.getItem('theme');
+  const [theme, setTheme] = useState<'dark' | 'light'>(storedTheme === 'light' ? 'light' : 'dark');
```
Comment (edichoska) [inline components/context/ThemeContext.tsx:14]: I think you will face issues here, because useState will initialize the theme before localStorage has the time to get the value from storage. 
Lets try adding the  `localStorage.getItem('theme')`; inside the useState and  then you will r`eturn storedTheme === 'light' ? 'light' : 'dark'`

Comment (DimitarHristovski) [reply components/context/ThemeContext.tsx:14]: Ok i changed the code.

Comment (edichoska) [reply components/context/ThemeContext.tsx]: @DimitarHristovski awesome! 🚀  That fixed the issue of returning to the default value when reloading.

What I am noticing now:
- When the toggle is on the left(the sun side) it should be the light theme. Right now they are mixed. 
**Right side=moon side is the dark theme(the default one). Left side=sun side is the light theme.**

-Another issue we are facing is with the logo. When the user toggles the circle to the left the logo blends with the background because the word Learnhub is white like the background. 
Could you change the logo/ logo letters to be dark when we are in light theme?
Thank you 🙏

Comment (DimitarHristovski) [reply components/context/ThemeContext.tsx]: The logo is a image that was provided before was text... i thought someone will give me the dark logo image.And about the toggle i was just following the design.If its on the right the will be dark and on the left light.

File: components/reusable-components/navigation/Navigation.tsx [hunk 7]
```diff
@@ -1,36 +1,72 @@
...
+  }, []);
+  const handleClick = () => {
+    toggleTheme();
+    setIsSun((prevIsSun) => {
+      const newIsSun = !prevIsSun;
+      localStorage.setItem('isSun', String(newIsSun));
+      return newIsSun;
+    });
+  };
+
```
Comment (edichoska) [inline components/reusable-components/navigation/Navigation.tsx:40]: Since `isSun` is true for light theme and false for dark theme, we can eliminate the isSun state and we can derive `isSun` value directly from the theme.
This reduces the component's state complexity and keeps your UI in sync with the theme state.

- **handleClick function** should just toggle the theme. We should let the theme context handle everything, including localStorage.

- and we might need an useEffect here that would apply theme class to body (we maybe should move that to higher level)

File: components/reusable-components/navigation/Navigation.tsx [hunk 8]
```diff
@@ -1,36 +1,72 @@
...
-          </div>
+        <Link className={`display-s ${styles.navigationLogo}`} href="/">
+          <Image
+            src="/logo/Horizontal-Logo_Full-Color.svg"
+            className={`${styles.navigationLogo}`}
+            alt="LearnHub Logo"
+            width={250}
+            height={55}
+          />
+        </Link>
```
Comment (edichoska) [inline components/reusable-components/navigation/Navigation.tsx:52]: We should toggle the logo as well.
If the theme is dark we should display the logo with white letters.
If the theme is light we show the logo you have placed here.

**Comment by edichoska** [comment]
Screenshot 2024-04-08 at 21 13 24
3a8">
Can we center the icon horizontally?
Also in the dark section the icon goes outside of the container.
@DimitarHristovski

**Comment by edichoska** [comment]
We have created a new branch because this one got polluted in the revert commit.
The changes here are merged.
